wp/lib/bap_wp/src/environment.ml

```diff
@@ -495,7 +502,7 @@ let mk_init_var (env : t) (var : Var.t) : Constr.z3_expr * t =
   let ctx = get_context env in
   let z3_var, _ = get_var env var in
   let sort = Expr.get_sort z3_var in
-  let name = Format.sprintf "init_%s" (Expr.to_string z3_var) in
+  let name = Format.sprintf "init_%s" (String.strip ~drop:(fun c -> Char.(c = '|')) (Expr.to_string z3_var)) in
```

Comment:
Can you explain what this line is for?

Reply:
Z3 (and smtlib) has a quoting mechanism where if a name contains certain characters it will wrap the name in "|" for example init_#248 will be printed as |init_#248| because # is a special character. It does this when you print Expr, but not when you print the Symbols from Z3 or make them. Since I'm taking the names from Expr to build the symbols, I'm stripping off this "|" character. This does deserve a comment, so I was also confused what this was doing exactly when i came back to it.

Reply:
So what's happening is that the `|`'s cause things to break while using a different SMT solver when trying to create symbols based off of the expr strings?

Reply:
It's really that internally to Z3 the reader for Symbol and the printer for Expr are not compatible in the presence of quotation I believe. That sounds crazy. Maybe I should double check that.

Reply:
I think that makes sense. It might be a good idea to ping @gltrost about this since it seems like he is doing some handling of the `|`s found in our queries.

Reply:
Thanks for the ping. Yes, this is related to issue #300 I'm working on, and dealing with the `#`-vars as they cause escaped `\|`s will make printing more readable.